internal/component/prometheus/operator/types.go

```diff
@@ -54,6 +57,7 @@ var DefaultArguments = Arguments{
 	Client: kubernetes.ClientArguments{
 		HTTPClientConfig: config.DefaultHTTPClientConfig,
 	},
+	KubernetesRole: string(promk8s.RoleEndpoint),
```

Comment:
Sorry, one more thing: in `Validate()` in line 69 we should check that it's either `endpoint` or `endpointslice` - other values are not supported.